lib/Doctrine/DBAL/DriverManager.php

```diff
@@ -232,18 +245,25 @@ private static function parseDatabaseUrl(array $params)
         // (pdo_)?sqlite3?:///... => (pdo_)?sqlite3?://localhost/... or else the URL will be invalid
         $url = preg_replace('#^((?:pdo_)?sqlite3?):///#', '$1://localhost/', $params['url']);
 
-        $url = parse_url($url);
+        // PHP < 5.4.8 doesn't parse schemeless urls properly.
+        // See: https://php.net/parse-url#refsect1-function.parse-url-changelog
+        if (PHP_VERSION_ID < 50408 && strpos($url, '//') === 0) {
+            $url = parse_url('fake:' . $url);
+
+            unset($url['scheme']);
+        } else {
+            $url = parse_url($url);
+        }
 
         if ($url === false) {
             throw new DBALException('Malformed parameter "url".');
         }
 
-        if (isset($url['scheme'])) {
-            $params['driver'] = str_replace('-', '_', $url['scheme']); // URL schemes must not contain underscores, but dashes are ok
-            if (isset(self::$driverSchemeAliases[$params['driver']])) {
-                $params['driver'] = self::$driverSchemeAliases[$params['driver']]; // use alias like "postgres", else we just let checkParams decide later if the driver exists (for literal "pdo-pgsql" etc)
-            }
-        }
+        // If we have a connection URL, we have to unset the default PDO instance connection parameter (if any)
+        // as we cannot merge connection details from the URL into the PDO instance (URL takes precedence).
+        unset($params['pdo']);
```

Comment:
@KingCrunch yeah I get what you mean. TBH it also took me some time to gather what actually the expected behaviour is/should be. The thing is (and after thinking a little bit about it also makes a lot of sense) the documentation propagates that if a URL is specified, it's information takes precedence over "default" value being set already in the connection parameters. This makes sense because as a service provider you could for example provide connection details for a default connection while the client can fully or partially ovewrite those settings. I think this was also part of the idea of your original PR. Telling that, a "default" PDO makes no differences in semantics towards providing a `driver` or `driverClass` because all of those end up in a connection instance using default values. So IMO making an exception for PDO here is rather confusing and defeats the original message of an URL taking precedence. Also silently discarding a connection URL may be dangerous to the client as he might expect to be connected to somewhere else than the default PDO connection. So IMO when providing a URL we have to error as soon as we have connection parameter conflicts.